registry/quilt_server/views.py

```diff
@@ -778,6 +787,47 @@ def package_put(owner, package_name, package_hash):
                     dict(team=app.config['TEAM_ID'], user=owner, pkg=package_name)
                 )
 
+    if package_path is not None:
```

Comment:
Probably cleaner to implement this conditional in a separate function.

Reply:
Moving everything into a new function becomes kind of awkward: we need the `tag` variable later, so the function would need to return `(contents, tag)`. Also, locking the DB table as a side effect is weird. I'll move the node traversal logic for now.

Reply:
That's enough refactoring for this PR.